x/incentive/keeper/rewards.go

```diff
@@ -108,19 +184,202 @@ func (k Keeper) SynchronizeReward(ctx sdk.Context, cdp cdptypes.CDP) {
 	claim.RewardIndexes[index].RewardFactor = globalRewardFactor
 	newRewardsAmount := rewardsAccumulatedFactor.Mul(cdp.GetTotalPrincipal().Amount.ToDec()).RoundInt()
 	if newRewardsAmount.IsZero() {
-		k.SetClaim(ctx, claim)
+		k.SetUSDXMintingClaim(ctx, claim)
 		return
 	}
 	newRewardsCoin := sdk.NewCoin(types.USDXMintingRewardDenom, newRewardsAmount)
 	claim.Reward = claim.Reward.Add(newRewardsCoin)
-	k.SetClaim(ctx, claim)
+	k.SetUSDXMintingClaim(ctx, claim)
 	return
 }
 
+// InitializeHardSupplyReward initializes the supply-side of a hard liquidity provider claim
+// by creating the claim and setting the supply reward factor index
+func (k Keeper) InitializeHardSupplyReward(ctx sdk.Context, deposit hardtypes.Deposit) {
+	_, claimFound := k.GetHardLiquidityProviderClaim(ctx, deposit.Depositor)
+	if claimFound {
+		return
+	}
+
+	var supplyRewardIndexes types.RewardIndexes
+	for _, coin := range deposit.Amount {
+		_, found := k.GetHardSupplyRewardPeriod(ctx, coin.Denom)
+		if !found {
+			continue
+		}
+
+		supplyFactor, foundSupplyFactor := k.GetHardSupplyRewardFactor(ctx, coin.Denom)
+		if !foundSupplyFactor {
+			supplyFactor = sdk.ZeroDec()
+		}
+
+		supplyRewardIndexes = append(supplyRewardIndexes, types.NewRewardIndex(coin.Denom, supplyFactor))
+	}
+
+	claim, found := k.GetHardLiquidityProviderClaim(ctx, deposit.Depositor)
+	if !found {
+		claim = types.NewHardLiquidityProviderClaim(deposit.Depositor,
+			sdk.NewCoin(types.HardLiquidityRewardDenom, sdk.ZeroInt()),
+			supplyRewardIndexes, nil, nil)
+		k.SetHardLiquidityProviderClaim(ctx, claim)
+		return
+	}
+
+	// User has a HARD claim, but supply reward indexes are empty
+	claim.SupplyRewardIndexes = supplyRewardIndexes
+	k.SetHardLiquidityProviderClaim(ctx, claim)
+}
+
+// SynchronizeHardSupplyReward updates the claim object by adding any accumulated rewards
+// and updating the reward index value
+func (k Keeper) SynchronizeHardSupplyReward(ctx sdk.Context, deposit hardtypes.Deposit) {
+	claim, found := k.GetHardLiquidityProviderClaim(ctx, deposit.Depositor)
+	if !found {
+		return
+	}
+
+	for _, coin := range deposit.Amount {
+		supplyFactor, found := k.GetHardSupplyRewardFactor(ctx, coin.Denom)
+		if !found {
+			continue
+		}
+
+		supplyIndex, hasSupplyRewardIndex := claim.HasSupplyRewardIndex(coin.Denom)
+		if !hasSupplyRewardIndex {
+			continue
```

Comment:
Just as a note, I don't think this should ever happen. Might be worth adding a print statement or something here on testnet and seeing if it's ever hit.

Reply:
Added print log (with TODO so we remove it later) in https://github.com/Kava-Labs/kava/pull/757/commits/172ecc89fbea476da913136f65873d17a21de110